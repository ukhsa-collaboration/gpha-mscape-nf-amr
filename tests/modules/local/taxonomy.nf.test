nextflow_process {

    name "Test Process READ_ANALYSIS"
    script "modules/local/taxonomy.nf"
    process "READ_ANALYSIS"

    test("test_read-analysis_positive") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = = Channel.of(
                    tuple(
                        "climb-id", 
                        file("${projectDir}/tests/test_data/ERR3077524_GCF_013349085.1.test_data.kraken_pluspf_8gb.tsv", checkIfExists: true),
                        file("${projectDir}/tests/test_data/ERR3077524_GCF_013349085.1.test_data.kraken_pluspf_8gb.report.json", checkIfExists: true),
                        "card",
                        file("${projectDir}/tests/test_data/ERR3077524_GCF_013349085.1.test_data.abricate_card_out.tsv", checkIfExists: true)

                    )
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
