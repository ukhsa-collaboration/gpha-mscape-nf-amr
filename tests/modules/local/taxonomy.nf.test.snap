{
    "test_read-analysis_positive": {
        "content": [
            {
                "0": [
                    "climb-id_abricate_card_taxa_out.tsv:md5,06044970b57f90818aa13b9afbadec2b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-10-15T09:57:38.262874193"
    }
}